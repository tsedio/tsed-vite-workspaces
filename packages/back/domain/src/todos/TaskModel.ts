import { Default, Example, MaxLength, MinLength, Required } from "@tsed/schema";

export class TaskModel {
  @Required()
  @MinLength(3)
  @MaxLength(200)
  @Example("My task title")
  title: string;

  @Required()
  @Default(false)
  done: boolean = false;
}
