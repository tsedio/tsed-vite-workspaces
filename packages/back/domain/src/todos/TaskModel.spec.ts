import { PlatformTest } from "@tsed/common";
import { getJsonSchema } from "@tsed/schema";

import { TaskModel } from "./TaskModel";

describe("TaskModel", () => {
  beforeEach(PlatformTest.create);
  afterEach(PlatformTest.reset);

  it("generate the expected schema", () => {
    const schema = getJsonSchema(TaskModel);

    expect(schema).toMatchSnapshot();
  });
});
