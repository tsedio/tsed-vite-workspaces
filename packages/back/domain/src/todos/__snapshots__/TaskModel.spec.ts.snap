// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskModel generate the expected schema 1`] = `
Object {
  "properties": Object {
    "done": Object {
      "default": false,
      "type": "boolean",
    },
    "title": Object {
      "examples": Array [
        "My task title",
      ],
      "maxLength": 200,
      "minLength": 3,
      "type": "string",
    },
  },
  "required": Array [
    "title",
    "done",
  ],
  "type": "object",
}
`;
